### PR TITLE
Simple map lister implementation (concommand: maps) & Small vs fixes

### PR DESCRIPTION
**Pull request by fuzun**
Added a temporary map lister function until real one gets developed. (This is pretty decent for now.)
It is not very good to put it in filesystem.c but it uses some private(static) filesystem functions.

Screenshot: https://vgy.me/yme5X3.png

&&&

Some vs warning fixes, should not bother anyone.

### CODE REVIEW

**Comment by a1batross** [comment]
Try to use FS_Search.

**Comment by fuzun** [comment]
FS_Search tries to look everywhere in gamedir? I saw it but did not use because it seem to be heavier at first.

File: engine/common/filesystem.c [hunk 1]
```diff
@@ -3687,3 +3687,72 @@ static byte *W_LoadFile( const char *path, fs_offset_t *lumpsizeptr, qboolean ga
...
+		for ( ; listIndex != mapList.numstrings; ++listIndex )
+		{
+			Q_strncpy( mapName, mapList.strings[ listIndex ], sizeof( mapName ) - 1 );
+			mapName[ 255 ] = '\0';
+			if ( Q_stristr( mapName, ".bsp" ) == NULL )
+				continue;
+			++mapCount;
+			Msg( "%s\n", mapName );
+		}
+		Msg( "%s\nGamedir: %s - Map count: %d\n", seperator, GI->basedir, mapCount, mapCount );
```
Comment (fuzun) [inline engine/common/filesystem.c]: Needless parameter, will be removed.

**Comment by mittorn** [comment]
```
FS_Search( va( "maps/*%s*.bsp", Cmd_Argv( 1 ) ) );
```

**Comment by fuzun** [comment]
@a1batross @mittorn

**Comment by mittorn** [comment]
Q_strncpy and other string functions in xash3d always adds '\0' to the end of string

**Comment by fuzun** [comment]
@mittorn Can you check the latest commit? Modify if you want and decide for implementation as it is becoming rotten.

I don't know but is there a similar command? Maybe an alias can be made.

**Comment by a1batross** [comment]
I think this can be merged.

**Pull request merged**